File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.go
func countBadPairs(nums []int) int64 {
    freq := make(map[int]int)
    goodPairs := 0
    n := len(nums)

    for i, num := range nums {
        key := num - i
        if val, exists := freq[key]; exists {
            goodPairs += val
        }
        freq[key]++
    }

    // var badPairs int64 = 

    return int64(n*(n-1)/2 - goodPairs)
}

File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.ts
function countBadPairs(nums: number[]): number {
    let freq: Map<number, number> = new Map()
    let goodPairs = 0
    let n = nums.length

    for (let i = 0; i < n; i++) {
        const key = nums[i] - i
        goodPairs += freq.get(key) || 0
        freq.set(key, (freq.get(key) || 0) + 1)
    }

    const totalPairs = (n * (n - 1)) / 2
    const badPairs = totalPairs - goodPairs

    return badPairs;
};